optaplanner-core/src/main/java/org/optaplanner/core/impl/score/DefaultScoreExplanation.java

```diff
@@ -25,16 +25,18 @@
 import org.optaplanner.core.api.score.constraint.ConstraintMatchTotal;
 import org.optaplanner.core.api.score.constraint.Indictment;
 
-public final class DefaultScoreExplanation<Solution_> implements ScoreExplanation<Solution_> {
+public final class DefaultScoreExplanation<Solution_, Score_ extends Score<Score_>>
```

Comment:
This is a nice example of where the generification actually helps.

We are now bound to `<Score_>` and this makes it crystal clear that the entire code is supposed to work with instances of the same type.

Generalization is good. More generalization is better.

Reply:
I agree that here it's a good thing, no doubt about that :)